### ISSUE TITLE
Specifying the default rounds parameter returns an invalid hash

### ISSUE THREAD

**Issue by sedrubal** [besser82/libxcrypt#16]
Since commit 15d1e3bf3d259ea262f3d34e8c225a3abfabaa09 it is not possible to specify `rounds=5000` as part of a SHA512 salt any more. This is imho a huge breaking change, that should be documented in `NEWS` and in `crypt(5)`. I think this even violates the regex mentioned in `crypt(5)`.

Apart from this, I think it should definitively be allowed to request the default number of rounds. If you ever decide to increase this parameter, every hash *not* specifying it will be unusable. This exception is just needless overhead on both sides, libxcrypt and any software using it.

In the commit message you write:

> This is in keeping with the observation, in the Passlib documentation, that allowing more than one valid crypt output string for any given (rounds, salt, phrase) triple is asking for trouble.

If this is really a problem, then I'd recommend not allowing salts that do *not* request a rounds parameter. But this would break even more systems I think.

**Comment by sedrubal** [comment]
Thank you very much for the clarification. :+1: 

To explain my use case: I built a custom hasher for Django to use SHA512 hashes (for compatibility with other systems using the same database). Django expects that generating salts and hashing pass phrases is split into 2 steps (2 functions). I wanted to be sure that it uses a fix number of rounds and I also tried  to write the code as declarative as possible to make it easy for other people to understand the code. They should directly see how many rounds are used when they see the generated salts / hashes. As Python before 3.7 did not expose the rounds parameter for `crypt.mksalt` (I assume that this uses `crypt_gensalt` behind the scenes), I inserted `$rounds={rounds}` (in this case `$rounds=5000` into the generated salts. They will be passed to `crypt.crypt` in the `hash()` function. Everything worked as expected until some days ago and suddenly it was broken and I didn't find any docs that explained the issue.

After figuring out the reason I can migrate all hashes to strip out `$rounds=5000` and change the code in such way, that it doesn't insert `$rounds={rounds}` when rounds == 5000, but I think, this is quite intransparent and I think, that I'm not the only one who is affected by this API change.

So I think for better interoperability this change should be reverted and if a hypothetical "system B" appears, they should fix their libs to accept the default value of rounds, too :wink: If you decide, that you don't want to revert it, this behaviour should be well documented...
